pipeline-modules/deployment-service/aws/istio-eks-module.yaml

```diff
@@ -0,0 +1,459 @@
+          manageCertificate: {{manageCertificate}}
+          {% if istioInjection %}
+          istioInjection: enabled
+          {% else %}
+          istioInjection: disabled
+          {% endif %}
+          {% if deployment %}
+          deploy:
+            replicaCount: {{replicaCount}}
+            strategy: RollingUpdate
+            maxUnavail: {{maxUnavail}}
+            maxSurge: {{maxSurge}}
+            terminationGracePeriodSeconds: {{terminationGracePeriodSeconds}}
+            healthCheck:
+              liveFailThreshold: '{{liveFailThreshold}}'
+              liveSuccThreshold: '{{liveSuccThreshold}}'
+              readFailThreshold: '{{readFailThreshold}}'
+              readSuccThreshold: '{{readSuccThreshold}}'
+              delay: '{{probeDelay}}'
+              period: '{{probePeriod}}'
+              timeout: '{{probeTimeout}}'
+
+          image: #auto
+            repository: {{ repository }}
+            pullPolicy: Always
+            tag:  {{ tag }}
+            registryCredentials:
+              registry: $docker_registry
+              username: $docker_username" >> values.yaml
+
+          printf '    password: %s\n' "$(printf '%s' "$docker_password" | jq -aRs .)" >> values.yaml
+
+          echo "
+          {% endif %}
+          {% if ports %}
+          ports:
+          {% for port in ports%}
+            - name: {{port.name}}
+              containerPort: {{port.containerPort}}
+              protocol: {{port.protocol}}
+              {% if port.healthCheck %}
+              healthcheck: {{port.healthCheck}}
+              {% endif %}
+          {% endfor %}
+          {% endif %}
+
+          {% if ports %}
+          service:
+            type: {{ service_port }}
+            private: false
+          {% endif %}
+
+          {% if domains %}
+          domains:
+          {% for domain in domains %}
+          - name: {{domain.name}}
+          {% endfor %}
+          {% endif %}
+
+          {% if virtualService %}
+          virtualService:
+          {% for vs in virtualService %}
+            prefix: {{vs.prefix}}
+            rewrite:
+              enabled: {{vs.rewriteEnabled | lower }}
+              value: {{vs.rewriteValue}}
+          {% endfor %}
+          {% endif %}
+
+
+          autoscaling:
+            enabled: {{autoscalingEnabled}}
+            minReplicas: {{minReplicas}}
+            maxReplicas: {{maxReplicas}}
+            targetCPUUtilizationPercentage: {{targetCPUUtilizationPercentage}}
+
+          resourcequota:
+            quotaRequestsCpu: 100
+            quotaRequestsMem: 0.2
+            quotaLimitsCpu: 200
+            quotaLimitsMem: 0.3
+
+          resources:
+            limits:
+              cpu: {{cpuLimit}}
+              memory: {{memoryLimit}}
+            requests:
+              cpu: {{cpuRequest}}
+              memory: {{memoryRequest}}
+
+          lifecycle:
+            prestop:
+              command: ['/bin/sh', '-c', 'sleep 30']
```

Comment:
I've been meaning to change this. This command does not work currently when we kill the pod in alpine. I think we should try out just "sleep 30".

Reply:
Will test this & update